File: tests/fixtures/java/r002_log_without_redaction.java
// Fixture: should trigger R002-java (LLM I/O logged without redaction).
// Spring controller user input + Spring AI ChatClient response logged
// directly via SLF4J / java.util.logging / System.out, no redactor.
package com.example.vuln;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.ai.chat.client.ChatClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class LogLeaker {

    private static final Logger log = LoggerFactory.getLogger(LogLeaker.class);

    private final ChatClient chatClient;

    public LogLeaker(ChatClient chatClient) {
        this.chatClient = chatClient;
    }

    @PostMapping("/chat")
    public String chat(@RequestParam String prompt) {
        log.info("User asked: {}", prompt);  // R002-java

        String response = chatClient.prompt().user(prompt).call().content();
        log.info("Model returned: {}", response);  // R002-java

        log.debug("debug detail prompt={} response={}", prompt, response);  // R002-java

        System.out.println("console: prompt=" + prompt);  // R002-java

        log.warn("Audit: user submitted {}", prompt);  // R002-java

        return response;
    }
}

File: tests/fixtures/python/r002_log_without_redaction.py
"""Fixture: should trigger R002 (LLM I/O logged without redaction).

User input + LLM completion both flow into log statements with no
intermediate redactor / hasher / length-only projection. OWASP LLM02.

Suppresses DF001 + R001 (existing rules) by importing nemoguardrails
+ structlog so the golden cleanly shows R002 alone.
"""
import nemoguardrails  # noqa: F401  (suppresses DF001)
import structlog  # noqa: F401  (suppresses R001)

import logging

from flask import Flask, request
from langchain_openai import ChatOpenAI

app = Flask(__name__)
log = logging.getLogger(__name__)
chain = ChatOpenAI(model="gpt-4o-mini")


@app.route("/chat", methods=["POST"])
def chat():
    prompt = request.json["q"]
    log.info(f"User asked: {prompt}")  # R002

    response = chain.invoke(prompt)
    log.info(f"Model returned: {response.content}")  # R002

    log.debug("debug detail prompt=%s response=%s", prompt, response.content)  # R002

    print(f"console: prompt={prompt}")  # R002

    logging.warning(f"Audit: user submitted {prompt}")  # R002

    return response.content
